code/ProtectedT.h

```diff
@@ -36,8 +37,9 @@ struct protectedT
     {
         std::unique_lock<std::mutex> u_l(m);
         if (can_transform) {
-            auto f_t = mem_fn(f);
-            f_t(t, std::forward<Arg>(arg));
+            auto f_t = std::mem_fn(f);
+//            f_t(t, std::forward<Arg>(arg));
+            std::bind(f_t, t, std::forward<Arg>(arg));
```

Comment:
it is in fact auto f = std:bind(f, &t, std::forward<Arg>(arg)); and the mem_fn call is removed